### PR TITLE
Presentation attributes support

### PR DESCRIPTION
**Pull request by skjalgepalg**
## Why?
* Presently any attributes on the `<svg>`-tag are lost as part of how `@nrk/svg-to-js` generates new `<svg>` or `<symbol>` tags.
* Reducing the size of an svg by consolidating multiple identical presentation attributes from child-nodes to the `<svg>`-tag, is not possible with present version of `@nrk/svg-to-js`.
  * E.g. the `fill="currentColor"` attribute in the NrkDownload-icon disappears when processed by `@nrk/svg-to-js`. as noted in https://github.com/nrkno/core-icons/issues/181

## What's new?
* Add support for presentation attributes on `<svg>`-tags to be kept by default.
* Updated tests with some new cases to reflect the support of attributes in `<svg>`-tags
* Updated readme examples to reflect updated icons and functionality

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
+
+/**
+ *
+ * @param {String} markup Stringified svg markup
+ * @return {Object | null} Object with key-value pairs of attributes
```
Comment (AndersCan) [inline lib/utils.js:5]: Kanskje overkill, men dette liker jeg
```js
/**
 * @typedef {Object} MissingSvgStuff
 * @property {false} foundAttributes
 */
/**
 * @typedef {Object} SvgStuff
 * @property {true} foundAttributes
 * @property {string} svgAttrStr
 * @property {string} svgAttrStrJsx
 * @property {Record<keyof SVG_PRESENTATION_ATTRIBUTES, string | undefined>} (denne trengs vel egentlig ikke)
 */


// også
/**
 * ...
 * @return {MissingSvgStuff | SvgStuff}
 */
```
tror dette hadde funket 👌

File: lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    console.warn(`getSVGTagPresentationAttributes: Unable to find leading svg-tag, in markup: ${markup}`)
+    return null
+  }
+  // Select key and value from all attributes found in tagParsingPattern - ignores valueless attributes like `hidden`
+  const attributeParsingPattern = /(?:^|\s)([a-z-A-Z]*)\s*=\s*((?:'[^']*')|(?:"[^"]*")|\S+)/gi
+  const attrs = {}
+
+  for (let match; (match = attributeParsingPattern.exec(attributes));) {
+    const { 1: key, 2: val } = match
+    const isQuoted = val && (val[0] === '\'' || val[0] === '"')
```
Comment (AndersCan) [inline lib/utils.js:21]:
```suggestion
    const isQuoted = val && (val[0] === `'` || val[0] === `"`)
```

**Review by AndersCan — APPROVED** [review]
🔔 ❌ ⬇️

File: README.md [hunk 3]
```diff
@@ -83,7 +85,7 @@ const customOutputs = [{
 }]
 
 const options = {
-  banner: '/** Made with @nrk/svg-to-js **/',
+  banner: 'Made with @nrk/svg-to-js',
```
[the remark this answers is not in the document]
Comment (skjalgepalg) [reply README.md:88]: Innholdet legges inn i utkommentert kontekst f.eks [her](https://github.com/nrkno/svg-to-js/blob/9bfa658ec2630ce0427565b62bf5f31a3f8d75d4/lib/index.js#L60), så å legge inn en kommentar her gir dobbel notasjon.

File: lib/utils.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+ * @param {string} markup Stringified svg markup
+ * @return {MissingSvgPresentationAttributes | SvgPresentationAttributes}
+ */
+export function parseSVGTagPresentationAttributes (markup) {
+  // Select tagName and all attributes as string from first matching tag
+  const tagParsingPattern = /<([a-z][a-z0-9]*)\s*([^>]*?)(\/?)>/
+  const { 1: tagName, 2: attributes } = tagParsingPattern.exec(markup)
+  if (tagName !== 'svg') {
+    console.warn(`getSVGTagPresentationAttributes: Unable to find leading svg-tag, in markup: ${markup}`)
+    return null
```
Comment (AndersCan) [inline lib/utils.js:28]:
```js
 return {
      foundAttributes: false
    }
```

File: lib/index.js [hunk 5]
```diff
@@ -29,6 +31,9 @@ export default function svgToJS (config) {
     const code = fs.readFileSync(path.join(config.input, file), 'utf-8')
     const size = String(code.match(/viewBox="[^"]+/)).slice(9) || `0 0 ${String(code.match(/width="[^"]+/)).slice(7)} ${String(code.match(/width="[^"]+/)).slice(7)}`
     const name = file.slice(0, -4)
+    const attrInfo = parseSVGTagPresentationAttributes(code)
+    const svgAttrStr = attrInfo.foundAttributes ? attrInfo.svgAttrStr : ''
+    const svgAttrStrJsx = attrInfo.foundAttributes ? attrInfo.svgAttrStrJsx : ''
```
Comment (AndersCan) [inline lib/index.js:36]: Dette syntes jeg ble bra 🎉

**Review by AndersCan — APPROVED** [review]
fikse `null` return så er dette 👌

**Review by eirikbacker — APPROVED** [review]

**Pull request merged**